solutions/apio-10-Commando.mdx

```diff
@@ -0,0 +1,112 @@
+---
+id: apio-10-Commando
+source: APIO 2010
+title: Commando
+author: Dustin Miao
+---
+
+# Explanation
+
+Let $\texttt{dp}[i]$ denote the maximum effectiveness of the prefix of length $i$ of the unit. The transitions are defined as $$\texttt{dp}[i] = \max_{1 \leq j < i} \{\texttt{dp}[j] + ax^2+bx+c\}$$, where $x = \texttt{sum}(j + 1, i) = \sum_{k = j + 1}^i x_k$. Define a prefix sum array $\texttt{pre} = \sum_{k = 1}^i x_k$. We can then express $x$ as $\texttt{pre}(i) - \texttt{pre}(j)$. 
```

Comment:
```suggestion
Let $\texttt{dp}[i]$ denote the maximum effectiveness of the prefix of length $i$ of the unit. The transitions are defined as $$\texttt{dp}[i] = \max_{0 \leq j < i} \{\texttt{dp}[j] + ax^2+bx+c\}$$, where $x = \texttt{sum}(j + 1, i) = \sum_{k = j + 1}^i x_k$. Define a prefix sum array $\texttt{pre}[i] = \sum_{k = 1}^i x_k$. We can then express $x$ as $\texttt{pre}(i) - \texttt{pre}(j)$. 
```

Reply:
huh why do you start with [] and then end up using ()

Reply:
shouldn't you start at 0 everywhere

Reply:
Why was this resolved?

Reply:
my mistake, forgot to commit it